lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -88,7 +91,13 @@ def define_virtual_aggregate_method(name, relation, method_name, column)
           if attribute_present?(name)
             self[name]
           elsif (rel = send(relation)).loaded?
-            rel.blank? ? nil : rel.map { |t| t.send(column) }.compact.send(method_name)
+            values = rel.map { |t| t.send(column) }.compact
+            return nil if values.blank?
+            if method_name == :average
```

Comment:
this is good.

I always had in my mind that we would pass a block into this method and `yield values` here but that is probably overkill